packages/web-components/fast-components-msft/src/badge/index.ts

```diff
@@ -2,17 +2,20 @@ import { attr, customElement } from "@microsoft/fast-element";
 import { Badge, BadgeTemplate as template } from "@microsoft/fast-foundation";
 import { BadgeStyles as styles } from "./badge.styles";
 
-export type BadgeFill = "accent" | "lightweight" | "neutral" | "string";
+export type BadgeAppearance = "accent" | "lightweight" | "neutral" | string;
 
 @customElement({
     name: "fast-badge",
     template,
     styles,
 })
 export class FASTBadge extends Badge {
-    @attr
-    public fill: BadgeFill = "lightweight";
-    private fillChanged(oldValue: BadgeFill, newValue: BadgeFill): void {
+    @attr({ mode: "fromView" })
+    public appearance: BadgeAppearance = "lightweight";
+    private appearanceChanged(
+        oldValue: BadgeAppearance,
+        newValue: BadgeAppearance
+    ): void {
         if (oldValue !== newValue) {
```

Comment:
This class should be applied through the template or this can cause a runtime error when this is set during construction

Reply:
Are there alternatives? This isn't global across all badges and ideally we'd reuse the template.

Reply:
Leveraging DOM.queueUpdate() here